File: PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7Ruby/lib/ejemplodsecuencia/sala.rb
class Sala
  
  attr_reader :aforo
  
  def initialize(a)
    @plazas_reservadas= Array.new
    @aforo=a
  end
  
  def reservaPlazas(nombre_cliente, fecha, numero_entradas)
        
        for i in 1..numero_entradas do #bucle loop
            
            #envío de mensaje 2.4.1 ,  crear es new, r es una variable local
            r= PlazaReservada.new(nombre_cliente, fecha)
            
            #envío de mensaje 2.4.2, incluir es <<, plazas reservadas es un atributo (ver diagrama de clases)
            @plazas_reservadas<<r 
            
        end
        
        
  end
    
    def obtener_numero_reservas(fecha)
      
    end
end

File: C1/PDOO/Material Complementario/Tema7/ejemplosTema7/ImplementacionEjemplosTema7/src/ejemplodsecuencia/Sala.java

package ejemplodsecuencia;

//solo se implementa aquello directamente relacionado con el diagrama de secuencia del ejemplo

import java.util.ArrayList;



class Sala {
    
    ArrayList <PlazaReservada> plazasReservadas;  //atributo de referencia, decido qué nombre ponerle
    int aforo;
    
    public void reservaPlazas(String nombreCliente, Date fecha, int numeroEntradas){
        
        for( int i=1; i<=numeroEntradas; i++){ //bucle loop
            
            //envío de mensaje 2.4.1 ,  crear es new, r es una variable local
            PlazaReservada r= new PlazaReservada(nombreCliente, fecha); 
            
            //envío de mensaje 2.4.2, incluir es add, plazas reservadas es un atributo (ver diagrama de clases)
            plazasReservadas.add(r); 
            
        }
        
        
    }
    
    public int getAforo(){
        
    }
    
    public int obtenerNumeroReservas(Date fecha){
        
    }
    
}
